framework/materials/multi_group_xs/openmc_import.cc

```diff
@@ -0,0 +1,197 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#include "framework/materials/multi_group_xs/multi_group_xs.h"
+#include "framework/logging/log.h"
+#include "framework/utils/hdf_utils.h"
+#include <numeric>
+#include <algorithm>
+
+namespace opensn
+{
+
+void
+MultiGroupXS::Initialize(const std::string& file_name,
+                         const std::string& dataset_name,
+                         double temperature)
+{
+  Reset();
+
+  // Disable internal HDF error reporting
+  H5Eset_auto2(H5E_DEFAULT, NULL, NULL);
+
+  // Open file
+  try
+  {
+    H5::H5File::isHdf5(file_name.c_str());
+  }
+  catch (H5::FileIException(&err))
+  {
+    std::string err_msg = file_name + " is not an HDF5 file.\n";
+    throw std::logic_error(err_msg);
+  }
+  H5::H5File file(file_name.c_str(), H5F_ACC_RDONLY);
+
+  // Check file type
+  std::string filetype;
+  H5ReadAttribute<std::string>(file, "filetype", filetype);
+  if (filetype != "mgxs")
+    throw std::runtime_error(file_name + " is not a valid OpenMC library file");
+
+  log.Log() << "Reading OpenMC cross-section file \"" << file_name << "\"\n";
+
+  // Number of groups
+  if (not H5ReadAttribute<size_t>(file, "energy_groups", num_groups_))
+    throw std::runtime_error("Failure reading \"energy_groups\" from " + file_name);
+
+  // Group structure
+  e_bounds_ = H5ReadDataset1D<double>(file, "/group structure");
+  std::reverse(e_bounds_.begin(), e_bounds_.end());
+
+  // Temperature
+  log.Log0() << file_name + " cross-section data evaluated at " << temperature << "K\n";
+  temperature_ = temperature;
+
+  // Base path
+  std::stringstream ss;
+  ss << std::fixed << std::setprecision(0) << temperature_ << "K";
+  std::string path = "/" + dataset_name + "/" + ss.str() + "/";
+  if (!H5Has(file, path))
+  {
+    throw std::runtime_error("Could not find dataset " + dataset_name + "/" + ss.str() + " in " +
+                             file_name);
+  }
+  temperature_ = temperature;
+
+  // Scattering order
```

Comment:
You have multiple of these blocks starting with the comment, but I can infer from the code and variable names what the block does, so the comment seems unnecessary, but since @ragusa and @zhardy-lanl did not comment on this, I assume this is somewhat valuable for you guys. I personally, would not have these comments...

Reply:
I personally find them useful. They let me easily scan the code and jump to the section where a specific process is being handled (especially with a good highlighting editor/ide).